sui-execution/latest/sui-move-natives/src/object_runtime/mod.rs

```diff
@@ -557,6 +557,22 @@ impl ObjectRuntimeState {
             })
             .collect::<Vec<_>>();
         for id in remaining_by_value_objects {
+            if input_objects
+                .get(&id)
+                .is_some_and(|owner| owner.is_shared())
+            {
+                return Err(ExecutionError::new(
+                    ExecutionErrorKind::SharedObjectOperationNotAllowed,
+                    Some(
+                        format!(
+                            "Internal invariant violation: Wrapping shared object {}",
```

Comment:
Invariant violation means that this is something that a user should not be able to trigger under normal operation, usually because we have some earlier check that prevents it.  Is that the case here? If not, we shouldn't describe this as an invariant violation.